### PR TITLE
CCITCARBON-138 Need to throw appropriate error messages when Validating the SDF yml file

### PR DESCRIPTION
**Pull request by shay6**

### CODE REVIEW

File: teflo/helpers.py [hunk 1]
```diff
@@ -1478,8 +1478,9 @@ def validate_cli_scenario_option(ctx, scenario, vars_data=None):
     try:
         scenario_stream = validate_render_scenario(scenario, vars_data)
         return scenario_stream
-    except yaml.YAMLError:
+    except yaml.YAMLError as err:
         click.echo('Error loading updated scenario data!')
+        click.echo(err.problem_mark)
```
Comment (rujutashinde) [inline teflo/helpers.py:1483]: make it as `click.echo(Error loading scenario file %s, % err.problem_mark)`
There is no need for two echo statements

**Review by rujutashinde — COMMENTED** [review]
Did you test this out using a scenario file which has included scenario and the included scenario has errors?
Please test this out and make sure you can catch errors with included scenarios as well

File: teflo/helpers.py (hunk 1 above)
Comment (shay6) [reply teflo/helpers.py]: done

File: teflo/helpers.py [hunk 2]
```diff
@@ -1330,9 +1330,9 @@ def validate_render_scenario(scenario, temp_data=None):
                         try:
                             yaml.safe_load(template_render(item, temp_data))
                             include_template.append(template_render(item, temp_data))
-                        except yaml.YAMLError:
+                        except yaml.YAMLError as err:
                             # raising Teflo error to differentiate the yaml issue is with included scenario
-                            raise TefloError('Error loading updated included scenario data!')
+                            raise TefloError('Error loading updated included scenario data! %s, %s!' % (item, err.problem_mark))
```
Comment (rujutashinde) [inline teflo/helpers.py:1335]: raise TefloError('Error loading the included scenario data! %s, %s!' % (item, err.problem_mark))
Remove the word updated, it is ambiguous as we are not really updating anything. it was incorrectly put before

**Review by rujutashinde — COMMENTED** [review]
squash the commits

File: teflo/helpers.py [hunk 3]
```diff
@@ -1478,15 +1478,15 @@ def validate_cli_scenario_option(ctx, scenario, vars_data=None):
     try:
         scenario_stream = validate_render_scenario(scenario, vars_data)
         return scenario_stream
-    except yaml.YAMLError:
-        click.echo('Error loading updated scenario data!')
+    except yaml.YAMLError as err:
+        click.echo('Error loading scenario file %s' % err.problem_mark)
```
Comment (rujutashinde) [inline teflo/helpers.py:1482]: problem_mark does not seem to be a property of err. I think you can just print the err it still will print the line number as to where the issue would be

File: teflo/helpers.py (hunk 2 above)
Comment (shay6) [reply teflo/helpers.py:1335]: done

File: teflo/helpers.py (hunk 3 above)
Comment (shay6) [reply teflo/helpers.py]: done

**Review by rujutashinde — APPROVED** [review]
LGTM

**Pull request merged**